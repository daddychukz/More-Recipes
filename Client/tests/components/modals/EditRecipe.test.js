import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import EditRecipe from '../../../src/components/modals/EditRecipeModal';


const props = {
  title: '',
  onChange: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  publicId: '',
  description: '',
  onSubmit: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  uploadWidget: sinon.spy(() => new Promise((cb) => {
    cb();
  }))
};

const shallowWrapper = shallow(<EditRecipe {...props} />);

describe('Edit Recipe Modal test suite', () => {
  it('renders edit recipe modal without crashing', () => {
    expect(shallowWrapper).toBeDefined();
  });

  it('should match component snapshot', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
