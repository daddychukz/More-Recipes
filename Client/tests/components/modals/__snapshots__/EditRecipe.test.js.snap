// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Recipe Modal test suite should match component snapshot 1`] = `
<div
  className="modal fade text-dark"
  id="editRecipe"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header bg-info"
      >
        <h5
          className="modal-title"
          id="contactModalTitle"
          style={
            Object {
              "color": "rgb(255, 255, 255)",
            }
          }
        >
          Edit Recipe
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="Title"
            >
              Recipe Title
            </label>
            <input
              className="form-control"
              name="title"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="Description"
            >
              Recipe Description
            </label>
            <textarea
              className="form-control"
              name="description"
              onChange={[Function]}
              required={true}
              rows="5"
              type="address"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="imgFile"
            >
              Upload Image
            </label>
            <CloudinaryContext
              cloudName="undefined"
            >
              <Image
                publicId=""
              >
                <Transformation
                  crop="scale"
                  dpr="auto"
                  height="100"
                  responsive_placeholder="blank"
                  width="100"
                />
              </Image>
            </CloudinaryContext>
            <input
              className="form-control-file"
              onClick={[Function]}
              type="button"
              value="Upload Image"
            />
            <small
              className="form-text text-muted"
              id="fileHelp"
            >
              Please upload an example image of recipe for better reviews.
            </small>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-info"
              data-dismiss="modal"
              onClick={[Function]}
              type="submit"
            >
              Update
            </button>
            <button
              className="btn btn-info"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
